tools/test_utils/fullservice.py

```diff
@@ -0,0 +1,215 @@
+        self.account_ids = None
+        self.request_count = 0
+        self.remove_wallet_and_ledger = remove_wallet_and_ledger
+        
+    def __enter__(self):
+        self.remove_wallet_and_ledger = True
+        self.start()
+        return self
+    
+    def __exit__(self,exc_type, exc_val, exc_tb):
+        self.stop()
+        if self.remove_wallet_and_ledger:
+            try:
+                print(f"Removing ledger/wallet dbs")
+                tmpdir = pathlib.Path('/tmp')
+                shutil.rmtree(tmpdir/'wallet-db')
+                shutil.rmtree(tmpdir/'ledger-db')
+            except Exception as e:
+                print(e) 
+        
+
+
+    def start(self):
+        cmd = ' '.join([
+            'mkdir -p /tmp/wallet-db/',
+            f'&& {constants.FULLSERVICE_DIR}/target/release/full-service',
+            '--wallet-db /tmp/wallet-db/wallet.db',
+            '--ledger-db /tmp/ledger-db/',
+            '--peer insecure-mc://localhost:3200',
+            '--peer insecure-mc://localhost:3201',
+            f'--tx-source-url file://{constants.MOBILECOIN_DIR}/target/release/mc-local-network/node-ledger-distribution-0',
+            f'--tx-source-url file://{constants.MOBILECOIN_DIR}/target/release/mc-local-network/node-ledger-distribution-1',
+        ])
+        print('===================================================')
+        print('starting full service')
+        print(cmd)
+        self.full_service_process = subprocess.Popen(cmd, shell=True)
+
+    def stop(self):
+        try:
+            subprocess.check_output("pkill full-service", shell=True)
+        except subprocess.CalledProcessError as exc:
+            if exc.returncode != 1:
+                raise
+
+    # return the result field of the request
+    def _request(self, request_data):
+        self.request_count += 1
+        print('sending request to full service')
+        url = 'http://127.0.0.1:9090/wallet'
+        default_params = {
+            "jsonrpc": "2.0",
+            "api_version": "2",
+            "id": self.request_count,
+        }
+        request_data = {**request_data, **default_params}
+
+        print(f'request data: {request_data}')
+
+        try:
+            parsed_url = urlparse(url)
+            connection = http.client.HTTPConnection(parsed_url.netloc)
+            connection.request('POST', parsed_url.path, json.dumps(request_data), {'Content-Type': 'application/json'})
+            r = connection.getresponse()
+        except ConnectionError:
+            raise ConnectionError
+
+        raw_response = None
+        try:
+            raw_response = r.read()
+            response_data = json.loads(raw_response)
+        except ValueError:
+            raise ValueError('API returned invalid JSON:', raw_response)
+        print(f'request returned {response_data}')
+
+        # TODO requests might be flakey due to timing issues... waiting 2 seconds to bypass most of these issues
+        time.sleep(2)
```

Comment:
I don't think it makes sense to always sleep 2 seconds after sending an HTTP request. What does that get you here?

Reply:
I think this is related to a FS bug actually. `http.client` expects connections to stay open per https://github.com/python/cpython/blob/3.10/Lib/http/client.py#L383 and FS doesn't seem to let this happen: 
<img width="1523" alt="image" src="https://user-images.githubusercontent.com/77425619/191104976-9658a9ed-c6c5-455b-92b8-74cd89d70440.png">
@BrianCorbin any ideas?

Reply:
The fact it says the connection was left intact means it didn't get closed, as expected.

Reply:
persistent connections are the default for HTTP 1.1, for example curling google.com twice gives me this 
<img width="479" alt="image" src="https://user-images.githubusercontent.com/77425619/191144897-82414e58-f5f1-41ff-a4a2-03a7ffdc315f.png">

Reply:
Ah, I see. The connection does die for some reason.

Reply:
https://github.com/mobilecoinofficial/full-service/pull/453

Maybe this 🤔

Reply:
This remains an issue. I don't see why we would need to sleep 2 seconds after every request we issue to FS, no matter what the response is.